### PR TITLE
Serve every html file under web

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Resolve any .html files to something under `web/`. This allows using multiple different html files besides index.html. Also adds a test case for serving and failing to serve one of these files.

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/web_fs.dart [hunk 1]
```diff
@@ -265,7 +265,16 @@ class AssetServer {
   Directory partFiles;
 
   Future<Response> handle(Request request) async {
-    if (request.url.path.contains('stack_trace_mapper')) {
+    if (request.url.path.endsWith('.html')) {
+      final Uri htmlUri = flutterProject.web.directory.uri.resolveUri(request.url);
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/web_fs.dart]: Should we place some basic protection in place for preventing serving files outside the project directory? Unless `resolveUri` already does it, of course.

File: packages/flutter_tools/lib/src/project.dart [hunk 2]
```diff
@@ -636,6 +636,9 @@ class WebProject {
   /// The 'lib' directory for the application.
   Directory get libDirectory => parent.directory.childDirectory('lib');
 
+  /// The directory containing additional html files for the application,
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/project.dart]: nit: could be any static files, no? Also, you probably meant to end with a period, not a comma.

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/build_runner/web_fs.dart [hunk 3]
```diff
@@ -265,7 +265,16 @@ class AssetServer {
   Directory partFiles;
 
   Future<Response> handle(Request request) async {
-    if (request.url.path.contains('stack_trace_mapper')) {
+    if (request.url.path.endsWith('.html')) {
```
Comment (DaveShuckerow) [inline packages/flutter_tools/lib/src/build_runner/web_fs.dart]: This is a complicated file handler.

Shelf also has static handlers that can serve all files out of a directory, which might make this logic a little simpler.

**Review by DaveShuckerow — COMMENTED** [review]
This may be useful!

File: packages/flutter_tools/lib/src/build_runner/web_fs.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_fs.dart]: You cannot send a relative uri so this should not be a problem regardless

File: packages/flutter_tools/lib/src/project.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/project.dart]: Fixed

File: packages/flutter_tools/lib/src/build_runner/web_fs.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/web_fs.dart]: Agreed, but as a long term goal I would like to remove our `shelf` dependency

**Pull request merged**